staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -866,6 +872,34 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 		}
 	}
 
+	if opts.requireMapListKeysMapSetValidation && schema.XListType != nil && (*schema.XListType == "map" || *schema.XListType == "set") && schema.Items != nil && schema.Items.Schema != nil {
+		is := schema.Items.Schema
+		if is.Nullable {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("items").Child("schema").Child("nullable"), "cannot be nullable when x-kubernetes-list-type is "+*schema.XListType))
+		}
+
+		if *schema.XListType == "map" {
+			requiredMap := make(map[string]struct{}, len(is.Required))
+			for _, required := range is.Required {
+				requiredMap[required] = struct{}{}
+			}
+
+			for _, k := range schema.XListMapKeys {
+				obj := is.Properties[k]
+
+				if _, required := requiredMap[k]; !required {
+					if obj.Default == nil {
+						allErrs = append(allErrs, field.Required(fldPath.Child("items").Child("schema").Child("properties").Key(k).Child("default"), "must be set if key is not required"))
```

Comment:
the `.schema` is just here in the Go types for OpenAPI. It's invisible in the OpenAPI schema itself. Just drop that piece in the path.